File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    int mem[] = new int[50];
    public int climbStairs(int n) {
        if (n <= 1 ){
            return 1;
        }
        if (mem[n-1] == 0) {
            mem[n-1] = climbStairs(n-1);
        } 
         if (mem[n-2] == 0) {
            mem[n-2] = climbStairs(n-2);
        }
        return mem[n-1]+mem[n-2];
    }
}

File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n < 3)
            return n;
        int twoStep = 2, oneStep = 1;
        while(n > 2)
        {
            twoStep += oneStep;
            oneStep = twoStep - oneStep;
            --n;
        }
        return twoStep;
    }
};